dev/DynamicDependency/MddLifetimeManagement.cpp

```diff
@@ -101,81 +101,90 @@ STDAPI MddLifetimeManagementGC() noexcept try
     winrt::Windows::Management::Deployment::PackageManager packageManager;
     winrt::hstring currentUser;
 
-    const UINT32 c_maxMajorVersion{ 9 };
+    const UINT16 c_majorMinorVersions[][3]{ // [n] = { Major, minMinor, maxMinor }
+        { 0, 8, 9 },
+        { 1, 0, 2 }
```

Comment:
is the removal process going to break older versions? E.g. when 1.3 releases, it will remove 1.2, but this package doesn't know to look for 1.3. Or am I misunderstanding this relationship?

Reply:
Nope. By definition we will only ever remove _unnecessary_ DDLMs, meaning if a DDLM is the only one to satisfy a need then it stays.

DDLM packages are named with their exact version, e.g. **Microsoft.ProjectReunion.DDLM-0.8.0.0-x64**_<version>_x64__8wekyb3d8bbwe. If the only 0.8 x64 DDLM registered is 0.8.0.0 then it stays. If you've got 0.8.0.1, 0.8.0.2 and 0.8.0.3 then .1 + .2 can be removed (if they're not in use) as .3 will meet all callers' needs.

That's why the logic here enumerates all the DDLMs with the same Major.Minor+Architecture and doesn't mess with the highest version